Tests/ARTFallbackTest.m

```diff
@@ -43,4 +43,35 @@ - (void)testAllHostsIncludedOnce {
     }    
 }
 
+- (void)testCustomFallbackHosts {
+    NSArray *customHosts = @[@"test.ably.com", @"test2.ably.com", @"test3.ably.com"];
+    ARTFallback *f = [[ARTFallback alloc] initWithFallbackHosts:customHosts];
+    
+    NSSet *customSet = [NSSet setWithArray:customHosts];
+    NSMutableArray *hostsRandomised = [NSMutableArray array];
+    for(int i=0;i < [customHosts count]; i++) {
+        [hostsRandomised addObject:[f popFallbackHost]];
+    }
+    
+    //popping after all hosts are exhausted returns nil
+    XCTAssertTrue([f popFallbackHost] == nil);
+    
+    // all fallback hosts are used in artfallback
+    XCTAssertEqual([hostsRandomised count], [customHosts count]);
+    bool inOrder = true;
+    for(int i=0;i < [customHosts count]; i++) {
+        if(![[customHosts objectAtIndex:i] isEqualToString:[hostsRandomised objectAtIndex:i]]) {
+            inOrder = false;
+            break;
+        }
+    }
+    //check artfallback randomises the order.
+    XCTAssertFalse(inOrder);
```

Comment:
BTW. This test will unfortunately fail every 6 (3x2) times as randomising a set of fallback hosts will be in the original order once every 6 times statistically.  So this test is probably not ideal.  We could make it statistically unlikely to fail by having a decent number of random hosts (12 would result in failure once every 480 million test runs) or simply run the tests multiple times in a row and only fail if all test runs fail.